src/test/scala/io/iohk/ethereum/blockchain/sync/regular/FetcherServiceSpec.scala

```diff
@@ -0,0 +1,75 @@
+package io.iohk.ethereum.blockchain.sync.regular
+
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.testkit.{TestKit, TestProbe}
+import akka.util.ByteString
+import cats.data.EitherT
+import io.iohk.ethereum.Fixtures.Blocks.ValidBlock
+import io.iohk.ethereum.Mocks.{MockValidatorsAlwaysSucceed, MockValidatorsFailingOnBlockBodies}
+import io.iohk.ethereum.blockchain.sync.PeerRequestHandler.RequestFailed
+import io.iohk.ethereum.blockchain.sync.TestSyncConfig
+import io.iohk.ethereum.consensus.validators.BlockValidator
+import io.iohk.ethereum.domain.{Block, BlockBody, BlockHeader, BlockchainReader}
+import io.iohk.ethereum.network.{Peer, PeerId}
+import io.iohk.ethereum.utils.Hex
+import monix.eval.Task
+import org.scalatest.PrivateMethodTester
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import org.scalamock.scalatest.MockFactory
+
+class FetcherServiceSpec extends TestKit(ActorSystem("FetcherServiceSpec_System")) with AnyFlatSpecLike with Matchers with PrivateMethodTester with MockFactory with TestSyncConfig {
+
+  val genesisHash: ByteString = ByteString(Hex.decode("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"))
+
+  //TODO: add a  happy path scenario once the implementation for fetching is finished
+  "FetcherService" should "return RequestFailed when asking for a not existing blocks" in {
+    val blockchainReader: BlockchainReader = mock[BlockchainReader]
+    (blockchainReader.getBlockHeaderByHash _).expects(*).returning(None).anyNumberOfTimes()
+    val blockValidator: BlockValidator = mock[BlockValidator]
+    val fetcherService = new FetcherService(blockValidator, blockchainReader, syncConfig)
+    val peerProbe: TestProbe = TestProbe()
+    val peer = Peer(PeerId("peerId"), new InetSocketAddress(9191), peerProbe.ref, true)
+    val result = fetcherService.fetchBlocksUntil(peer, Right(ByteString("byteString")), Right(ByteString("byteString")))
+    result.value shouldBe Task.now(Left(RequestFailed(peer, "Couldn't find blocks to fetch")))
+  }
+
+  //TODO: enable when the implementation for fetching is finished
+  it should "fetch blocks and return a peer or an error" ignore {
```

Comment:
Being this a test this description should not be "return this or that". I see this test returns an error, so like the previous test it would be more clear a description more like "return RequestFailed when <something>"